src/lib.rs

```diff
@@ -9,7 +9,7 @@ use ads1x1x::{
 };
 use ak09915_rs::{Ak09915, Mode as mag_Mode};
 use bmp280::{Bmp280, Bmp280Builder};
-use embedded_hal::prelude::_embedded_hal_blocking_delay_DelayMs;
+use embedded_hal::{digital::v2::InputPin, prelude::_embedded_hal_blocking_delay_DelayMs};
```

Comment:
undo

Reply:
Why ? This trait is in use.

Reply:
you updated it to use same spi/led pins traits, seems isn't required anymore,

Reply:
is_high is a function from InputPin trait, I'm not following you. As I said, the trait is required since we use a function from it.

Reply:
from:

`let leak = Pin::new(27);`

Or you mean that Pin is using this InputPin somehow?

Reply:
As I said, `is_high` is a function from the trait `embedded_hal::digital::v2::InputPin`.
You can check the documentation: https://docs.rs/embedded-hal/0.2.7/embedded_hal/digital/v2/trait.InputPin.html

I'm not sure how to be more clear. I'm not sure what you are not understanding. I see no point for repeating. If there is a clear way, please provide an example.

Reply:
k.